compiler/rustc_hir_typeck/src/demand.rs

```diff
@@ -827,27 +827,33 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         ) else {
             return;
         };
-        let in_scope_methods = self.probe_for_name_many(
-            probe::Mode::MethodCall,
-            path.ident,
-            Some(expected),
-            probe::IsSuggestion(true),
-            self_ty,
-            deref.hir_id,
-            probe::ProbeScope::TraitsInScope,
-        );
+        let in_scope_methods = self
+            .probe_for_name_many(
+                probe::Mode::MethodCall,
+                path.ident,
+                Some(expected),
+                probe::IsSuggestion(true),
+                self_ty,
+                deref.hir_id,
+                probe::ProbeScope::TraitsInScope,
+            )
+            .unwrap_or(Vec::new());
```

Comment:
Can you just use a `let Ok(in_scope_methods) = ... else { return }`

Reply:
Done